contracts/infinite-pool/RewardsPoolBaseInfinite.sol

```diff
@@ -0,0 +1,566 @@
+     */
+    function stake(uint256 _tokenAmount) public virtual {
+        _stake(_tokenAmount, msg.sender, true);
+    }
+
+    function _stake(
+        uint256 _tokenAmount,
+        address _staker,
+        bool _chargeStaker
+    ) internal {
+        uint256 currentTimestamp = block.timestamp;
+
+        require(
+            startTimestamp > 0 && currentTimestamp > startTimestamp,
+            'RewardsPoolBaseInfinite: staking is not started'
+        );
+
+        UserInfo storage user = userInfo[_staker];
+        require(
+            (user.amountStaked + _tokenAmount <= stakeLimit) && (totalStaked + _tokenAmount <= contractStakeLimit),
+            'RewardsPoolBaseInfinite: stake limit reached'
+        );
+
+        require(_tokenAmount > 0, 'RewardsPoolBaseInfinite: cannot stake 0');
+
+        // if no amount has been staked this is considered the initial stake
+        if (user.amountStaked == 0) {
+            user.firstStakedTimestamp = currentTimestamp;
+        }
+
+        updateRewardMultipliers(); // Update the accumulated multipliers for everyone
+        _updateUserAccruedReward(_staker); // Update the accrued reward for this specific user
+
+        user.amountStaked = user.amountStaked + _tokenAmount;
+        totalStaked = totalStaked + _tokenAmount;
+
+        uint256 rewardsTokensLength = rewardsTokens.length;
+
+        for (uint256 i = 0; i < rewardsTokensLength; i++) {
+            user.rewardDebt[i] = (user.amountStaked * accumulatedRewardMultiplier[i]) / PRECISION; // Update user reward debt for each token
+        }
+
+        emit Staked(_staker, _tokenAmount);
+
+        stakingToken.safeTransferFrom(address(_chargeStaker ? _staker : msg.sender), address(this), _tokenAmount);
+    }
+
+    /** @dev Claim all your rewards, this will not remove your stake
+     */
+    function claim() public virtual {
+        _claim(msg.sender);
+    }
+
+    function _claim(address _claimer) internal {
+        UserInfo storage user = userInfo[_claimer];
+        updateRewardMultipliers();
+        _updateUserAccruedReward(_claimer);
+
+        uint256 rewardsTokensLength = rewardsTokens.length;
+
+        for (uint256 i = 0; i < rewardsTokensLength; i++) {
+            uint256 reward = user.tokensOwed[i];
+            user.tokensOwed[i] = 0;
+            totalClaimed[i] = totalClaimed[i] + reward;
+
+            emit Claimed(_claimer, reward, rewardsTokens[i]);
+
+            IERC20(rewardsTokens[i]).safeTransfer(_claimer, reward);
+        }
+    }
+
+    /** @dev Withdrawing a portion or all of staked tokens. This will not claim your rewards
+     * @param _tokenAmount The amount to be withdrawn
+     */
+    function withdraw(uint256 _tokenAmount) public virtual {
+        _withdraw(_tokenAmount, msg.sender);
+    }
+
+    function _withdraw(uint256 _tokenAmount, address _withdrawer) internal {
+        require(_tokenAmount > 0, 'RewardsPoolBase: cannot withdraw 0');
+
```

Comment:
So this will be the fix in your opinion? 
`require(_tokenAmount <= user.amountStaked, 'RewardsPoolBaseInfinite: not enough funds to withdraw');`
In line 272

Reply:
I apply this, to the current commit, so take a look